File: 125.c
bool isPalindrome(char* s) {
    int size = 8;
    int index = 0;
    char* n = malloc(size * sizeof(char));

    while (*s) {
        if (isalnum(*s)) {
            while (index>= size) {
                size *= 2;
                n = realloc(n, size * sizeof(char));
            }
            n[index++] = tolower(*s);
        }
        s++;
    }

    for (int i = 0; i < index / 2; i++) {
        if (n[i] != n[index - 1 - i]) return 0;
    }
    return 1;
}

File: 125.py
class Solution:
    def isPalindrome(self, s: str) -> bool:
        lst = []
        for c in s:
            if c.isalnum():
                lst.append(c)
        s2 = "".join(lst)
        s2 = s2.lower()
        return s2 == s2[::-1]
